integration/query_test.go

```diff
@@ -1280,3 +1324,114 @@ func TestQueryCommandGetMore(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryCommandGetMoreConnection(t *testing.T) {
+	t.Parallel()
+
+	// set minPoolSize and maxPoolSize 1 to ensure only one pool exists duration of the test,
+	// which forces a client to use a single connection pool
+	s := setup.SetupWithOpts(t, &setup.SetupOpts{
+		ExtraOptions: url.Values{
+			"minPoolSize": []string{"1"},
+			"maxPoolSize": []string{"1"},
+		},
+	})
+
+	collection1 := s.Collection
+	databaseName := s.Collection.Database().Name()
+	collectionName := s.Collection.Name()
+
+	docs := generateDocuments(0, 5)
+	_, err := collection1.InsertMany(s.Ctx, docs)
+	require.NoError(t, err)
+
+	t.Run("SameClient", func(t *testing.T) {
+		// Do not run subtests in t.Parallel() to reduce the occurrence
+		// of session error. Even the same connection is used for the test
+		// by setting `minPoolSize` and `maxPoolSize` to 1, on rare occasion
+		// it returns error due to different session calling `find` and `getMore`.
+		// Supporting session would help us understand fix it
+		// https://github.com/FerretDB/FerretDB/issues/153.
+		//
+		// > Location50738
+		// > Cannot run getMore on cursor 2053655655200551971,
+		// > which was created in session 2926eea5-9775-41a3-a563-096969f1c7d5 - 47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU= -  - ,
+		// > in session 774d9ac6-b24a-4fd8-9874-f92ab1c9c8f5 - 47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU= -  -
+
+		var res bson.D
+		err = collection1.Database().RunCommand(s.Ctx,
+			bson.D{
+				{"find", collection1.Name()},
+				{"batchSize", 2},
+			}).Decode(&res)
+		require.NoError(t, err)
+
+		v, ok := res.Map()["cursor"]
+		require.True(t, ok)
+
+		cursor, ok := v.(bson.D)
+		require.True(t, ok)
+
+		cursorID := cursor.Map()["id"]
+		assert.NotNil(t, cursorID)
+
+		err = collection1.Database().RunCommand(s.Ctx, bson.D{
+			{"getMore", cursorID},
+			{"collection", collection1.Name()},
+		}).Decode(&res)
+		require.NoError(t, err)
+	})
+
+	t.Run("DifferentClient", func(t *testing.T) {
+		// error returned from using different clients are session related error,
+		// hence currently FerretDB does not return an error
+		setup.SkipExceptMongoDB(t, "https://github.com/FerretDB/FerretDB/issues/153")
+
+		t.Parallel()
```

Comment:
It also can't be a parallel test – it will be broken once we add another parallel subtest

Reply:
Yep thanks